plugins/scaffolder-backend/src/scaffolder/tasks/DatabaseTaskStore.ts

```diff
@@ -176,8 +179,19 @@ export class DatabaseTaskStore implements TaskStore {
     }
   }
 
-  private constructor(client: Knex) {
+  private constructor(client: Knex, events?: EventsService) {
     this.db = client;
+    this.events = events;
+  }
+
+  private getState(task: RawDbTaskRow) {
```

Comment:
I wonder if we need a special case for state 😅 
As same can apply for secrets too.
And the overall parsing of spec is already try-catched.

Have you faced some issue in particular with state?